### PR TITLE
Initial checkstyle checkin

### PR DESCRIPTION
**Pull request by SalusaSecondus**
*Issue #, if available:* #133

*Description of changes:*

Initial commit of checkstyle configuration. It is still just at warning level and hasn't been tuned for our code yet. (So, it's not enforcing.)

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: checkstyle.xml [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+  <module name="FileLength"/>
+  <module name="LineLength">
+    <property name="fileExtensions" value="java"/>
+    <property name="max" value="120" />
+    <property name="ignorePattern" value="^ *\* *[^ ]+$"/> <!-- Ignore javadoc of single long word. -->
+  </module>
+
+  <!-- Checks for whitespace                               -->
+  <!-- See https://checkstyle.org/config_whitespace.html -->
+  <module name="FileTabCharacter"/>
```
Comment (WillChilds-Klein) [inline checkstyle.xml:64]: ❤️

File: checkstyle.xml [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+    <property name="maximum" value="0"/>
+    <property name="message" value="Line has trailing spaces."/>
+  </module>
+
+  <!-- Checks for Headers                                -->
+  <!-- See https://checkstyle.org/config_header.html   -->
+  <!-- <module name="Header"> -->
+  <!--   <property name="headerFile" value="${checkstyle.header.file}"/> -->
+  <!--   <property name="fileExtensions" value="java"/> -->
+  <!-- </module> -->
```
Comment (WillChilds-Klein) [inline checkstyle.xml:80]: is this intentionally left commented out?

**Review by WillChilds-Klein — COMMENTED** [review]
LGTM, thanks for adding this!

**Review by WillChilds-Klein — APPROVED** [review]

File: checkstyle.xml (hunk 2 above)
Comment (SalusaSecondus) [reply checkstyle.xml:80]: I copy/pasted a base xml and it included this. We may change it later, so maybe?

**Pull request merged**